playbooks/roles/mongo_3_2/templates/backup-mongo.sh.j2

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# ref https://tasks.opencraft.com/browse/SE-1669
+# Script to perform a point-in-time dump of the local mongodb database using
+# mongodump.
+# includes locking (prevent this script from running multiple times in
+# parallel), creating a snapshot of the volume used to backup to, and snitch monitoring
+
+# vars
+ARCHIVE_DIR="TODO"
+ARCHIVE_NAME="mongo-$(date --iso-8601=minutes --utc).gz"
+MONGO_ADMIN_PASSWORD="TODO"
+VOLUME="TODO" # need some way to tell the script which is the volume being used - must be a dedicated ebs
+
+
+# TODO: lock
+
+# recreate the archive dir so we start afresh
+rm -rf "$ARCHIVE_DIR"
+mkdir -p "$ARCHIVE_DIR"
+
+
+# create the dump
+mongodump --host="localhost" --opLog --gzip --password "$MONGO_ADMIN_PASSWORD" --username "admin" --authenticationDatabase "admin" --archive="$ARCHIVE_NAME"
+
+# TODO: fsync the volume, create a snapshot, rotate old snapshots.
+#       alternately, upload to s3.
+
+
+# TODO: snitch/deadman's snitch here
+
+
+# TODO: save logs from this script?
```

Comment:
The cron that triggers this script should specify an output log file, which can be rotated using logrotate. 

* [ansible cron example](https://github.com/edx/configuration/blob/cf4d221d384ed396a3008c31487f385544ef08e7/playbooks/roles/logstash/tasks/main.yml#L58-L65)
* [logrotate template example](https://github.com/edx/configuration/blob/master/playbooks/roles/nginx/templates/etc/logrotate.d/edx_logrotate_nginx_access.j2)